File: backend_nestjs/src/restriction/entities/restriction.entity.ts
import { Animal } from 'src/animals/entities/animal.entity';
import {
  Entity,

  PrimaryGeneratedColumn,
  ManyToOne,
  JoinColumn
} from 'typeorm';

@Entity()
export class Restriction {
  @PrimaryGeneratedColumn()
  id: number;

  @ManyToOne(() => Animal, (animal) =>  animal.restrictions)
  @JoinColumn({name: "animal_id"})
  animal: Animal;

  @ManyToOne(() => Animal, (animal) =>  animal.restrictions)
  @JoinColumn({name: "animal_restricted_id"})
  animalRestrictedId: Animal;

}

File: BackendSpringboot/src/main/java/RIwi/BackendSpringboot/model/entities/RestrictionEntity.java
package RIwi.BackendSpringboot.model.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity(name = "Restriction")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RestrictionEntity {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "animal_id")
    private AnimalEntity animal;

    @ManyToOne
    @JoinColumn(name = "animal_restricted_id")
    private AnimalEntity animalRestringido;
}
